models/concerns/validate_data.rb

```diff
@@ -69,6 +74,25 @@ def ensure_no_conflicting_skills
     )
   end
 
+  def ensure_no_day_indices
+    day_indices =
+      # there can be no sequence_timewindows without schedule
+      @hash[:vehicles].flat_map{ |v| v[:timewindow].to_h[:day_index] } +
+      (@hash[:services].to_a + @hash[:shipments].to_a).flat_map{ |s|
+        activities = [s[:activity], s[:activities], s[:pickup], s[:delivery]].compact.flatten
+        activities.flat_map{ |a| a[:timewindows]&.flat_map{ |tw| tw[:day_index] } }
+      } +
+      @hash[:rests].to_a.flat_map{ |r| r[:timewindows].to_a.flat_map{ |tw| tw[:day_index] } } +
+      @hash[:timewindows].to_a.flat_map{ |tw| tw[:day_index] } +
+      @hash[:routes].to_a.flat_map{ |r| r[:day_index] ? r[:day_index] % 7 : nil }
+
+    return if day_indices.compact.uniq.size <= 1
+
+    raise OptimizerWrapper::DiscordantProblemError.new(
+      'There can not be different day indices if no schedule is provided'
```

Comment:
Both cannot and can not are perfectly fine, but cannot is far more common and is therefore recommended, especially in any kind of formal writing. https://www.merriam-webster.com/words-at-play/cannot-vs-can-not-is-there-a-difference